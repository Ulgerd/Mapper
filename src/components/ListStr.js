import React from 'react';
import Icon from './Icon.js';
import { Draggable } from "react-beautiful-dnd";
import styled from 'styled-components'

const StyledLi = styled.li`
   display: flex;
   align-items: center;
   margin-bottom: 0.5em;
`

const StyledSpan = styled.span`
  width: 90%;
  text-align: justify;
  padding-right: 1em;
`

export function ListStr(props) {
  return (
    <Draggable
      key={props.id}
      draggableId={props.id}
      index={props.index}
    >
      {(provided, snapshot) => (
        <StyledLi
          ref={provided.innerRef}
           {...provided.draggableProps}
           {...provided.dragHandleProps}
        >
          <StyledSpan>{props.index+1}. {props.name}</StyledSpan>
          <Icon
            onClick = {() => {props.onDotRemove(props.index)}}
            fill='black'
            xlink='close'
          />
        </StyledLi>
      )}
    </Draggable>
  );
}

export default ListStr;
