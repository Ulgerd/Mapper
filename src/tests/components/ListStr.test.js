import React from 'react';
import { shallow } from 'enzyme'
import {ListStr} from '../../components/ListStr.js';
import renderer from 'react-test-renderer';
jest.mock('../../components/Icon', () => 'Icon')

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) => children({
    draggableProps: {
      style: {},
    },
    innerRef: jest.fn(),
  }, {}),
  Draggable: ({ children }) => children({
    draggableProps: {
      style: {},
    },
    innerRef: jest.fn(),
  }, {}),
  DragDropContext: ({ children }) => children,
}));

describe('ListStr component tests',()=>{
    let wrapper

    const props = {
      key: 123,
      id: 123,
      index: 4,
      name: 'Moscow',
      onDotRemove: () => {console.log(1)}
    };

    beforeEach(()=>{
        wrapper = shallow(<ListStr {...props}/>)
    })

    it('render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });

    it('contains Draggable', () => {
      expect(wrapper.find('Draggable')).toHaveLength(1)
    });

    it('sets the svg element fill', () => {
      expect(wrapper.find('[draggableId=123]')).toHaveLength(1);
      expect(wrapper.find('[index=4]')).toHaveLength(1);
    });

    it('renders correctly', () => {
      const tree = renderer
        .create(<ListStr {...props} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});
