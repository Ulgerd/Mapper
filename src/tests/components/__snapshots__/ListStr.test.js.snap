// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListStr component tests renders correctly 1`] = `
<li
  className="sc-bdVaJa jgKfnK"
  style={Object {}}
>
  <span
    className="sc-bwzfXH efgKzg"
  >
    5
    . 
    Moscow
  </span>
  <Icon
    fill="black"
    onClick={[Function]}
    xlink="close"
  />
</li>
`;
